visualize-admin/visualization-tool
# Multiple hierarchy root handling

Handle multiple root hierarchy on the server by transforming the tree

```
Switzerland (hierarchy: cantons)
  Zurich, Appenzell, Schwiz....
Switzerland (hierarchy: production region)
  Jura, Plateau ...
Switzerland (hierarchy: protection region)
  Jura + Plateau, Southwestern alps
```

We notice that it is the same root (Switzerland) for the 3 hierarchies, so we transform the tree into

```
Switzerland (hierarchy: cantons)
  Cantons
    Zurich, Appenzell, Schwiz....
  Production region
    Jura, Plateau ...
  Protection region
    Jura + Plateau, Southwestern alps
```

This way, the front-end does not have to know about those multiple roots.

Putting this PR in draft as I'd like to see with Thomas Bettler if the hierarchy is correct as I see multiple hierarchy nodes with "-".

Should fix https://github.com/visualize-admin/visualization-tool/issues/936

How to test:

- Use this dataset : https://visualization-tool-git-feat-multiple-hierarchy-roots-ixt1.vercel.app/en/browse?dataset=https%3A%2F%2Fenvironment.ld.admin.ch%2Ffoen%2Fnfi%2FC-96-r%2Fcube%2F1&dataSource=Int
- Select NFI4 as inventory filter otherwise no data is shown on the chart (?)
- In the Region filter, you can see the multiple roots
